modules/kubernetes-addons/karpenter/data.tf

```diff
@@ -10,6 +10,7 @@ data "aws_iam_policy_document" "karpenter" {
       "ec2:RunInstances",
       "ec2:CreateTags",
       "iam:PassRole",
+      "ec2:DeleteLaunchTemplate*",
```

Comment:
Do you need `*` at the end?

Reply:
@vara-bonthu 
![image](https://user-images.githubusercontent.com/168366/168294531-803b8b5b-5cc4-4cd8-9e30-9913fdd61920.png)
since there will be updates to the templates, it generates versions of each, so I played it safe and added both permissions

Reply:
Right now Karpenter doesn't delete versions, just the launch template https://github.com/aws/karpenter/search?l=Go&q=DeleteLaunchTemplate

Reply:
Thanks @bryantbiggs 
@FernandoMiguel  Let's remove the `*` for now. We can add the other policies when its required

Reply:
@vara-bonthu @bryantbiggs updated

Reply:
Thank you sir, much appreciated!